src/modules/axis.js

```diff
@@ -1,26 +1,41 @@
 import i18n from '@dhis2/d2-i18n'
+import { AXIS_ID_COLUMNS, AXIS_ID_ROWS, AXIS_ID_FILTERS } from './layout/axis'
 import {
-    AXIS_ID_COLUMNS,
-    AXIS_ID_ROWS,
-    AXIS_ID_FILTERS,
-    AXIS_ID_YEAR_OVER_YEAR_SERIES,
-    AXIS_ID_YEAR_OVER_YEAR_CATEGORY,
-} from './layout/axis'
+    LAYOUT_TYPE_DEFAULT,
+    LAYOUT_TYPE_PIE,
+    LAYOUT_TYPE_YEAR_OVER_YEAR,
+    LAYOUT_TYPE_PIVOT_TABLE,
+} from './layoutTypes'
+import { getLayoutTypeByVisType } from './visTypeToLayoutType'
 
-const axisNames = {
-    [AXIS_ID_COLUMNS]: i18n.t('Series'),
-    [AXIS_ID_ROWS]: i18n.t('Category'),
-    [AXIS_ID_FILTERS]: i18n.t('Filter'),
-    [AXIS_ID_YEAR_OVER_YEAR_SERIES]: i18n.t('Series'),
-    [AXIS_ID_YEAR_OVER_YEAR_CATEGORY]: i18n.t('Category'),
+const getAxisNamesByLayoutType = layoutType => {
+    switch (layoutType) {
+        case LAYOUT_TYPE_DEFAULT:
+        case LAYOUT_TYPE_PIE:
+        case LAYOUT_TYPE_YEAR_OVER_YEAR:
+        default:
+            return {
+                [AXIS_ID_COLUMNS]: i18n.t('Series'),
```

Comment:
So this is actually a much bigger problem in this library... we **don't actually support i18n extraction and generation in @dhis2/analytics currently!**

We're using `d2-i18n` several places, even before this change, but since we don't actually look for those translations in the build step or pull them out into a `pot` file, let alone load them at runtime, they will never actually be translated.  CC @janhenrikoverland @edoardo @jenniferarnesen

Reply:
Good catch! This is a major issue. Continuing the discussion on Slack.

Reply:
Definitely something we need to address, but since it's out of scope for this specific PR I'm marking this as resolved.